MovingBlocks/DestinationSol
# Fix 309 incorrect number of systems when loading game

# Description
Fixes #309 
* introduced a WorldConfig class to be able to remove the static SYS_COUNT variable which was not updated when loading a save game.
* changed world file to be json and include the number of systems
* moved around constants and functions responsible for saving to uphold encapsulation and single responsibility principle
* tutorial is now started as 'new game' to be future proof

# Testing
1. Start a new game with 2 systems; open the map and memorize layout
2. Start a new game with 2 systems; open the map and verify the layout of the systems are different
3. Start a new game with 4 systems; open the map and memorize layout
4. Go back to the menu
5. Click on 'Continue' and verify there are 4 systems and the layout is the same
5. Exit and Restart the game; continue; verify there are 4 systems and the layout is the same
6. Start a new game with 3 planets; verify there are only 3 planets
7. Start a tutorial. The number of planets will probably be 3, but it does not matter. Memorize the layout
8. Start another tutorial. Verify that the layout is different
